usb/usb.c

```diff
@@ -142,20 +185,31 @@ static int usb_devsList(char *buffer, size_t size)
 
 static int usb_handleConnect(msg_t *msg, usb_connect_t *c)
 {
-	usb_drv_t *drv;
+	usb_drvpriv_t *drv;
 
-	if ((drv = malloc(sizeof(usb_drv_t))) == NULL)
+	drv = malloc(sizeof(usb_drvpriv_t));
+	if (drv == NULL) {
 		return -ENOMEM;
+	}
 
-	if ((drv->filters = malloc(sizeof(usb_device_id_t) * c->nfilters)) == NULL) {
+	drv->driver = malloc(sizeof(usb_driver_t));
```

Comment:
Is `usb_driver_t` always present (allocated) when `usb_drvpriv_t` is allocated? Maybe you can omit one `malloc()` and just statically allocate this structure inside `usb_drvpriv_t`?